developer/src/tike/child/Keyman.Developer.UI.Debug.UfrmLdmlKeyboardDebug.pas

```diff
@@ -474,39 +474,50 @@     TMemoSelectionState = record
     end;
   end;
 
+
+  // Returns the length of the context items in UTF16, treating markers as a
+  // single UTF16 code unit (U+FFFC)
+  function ContextItemsLengthUTF16(pc: pkm_core_context_item): Integer;
+  begin
+    Result := 0;
+    while pc._type <> KM_CORE_CT_END do
+    begin
+      if pc._type = KM_CORE_CT_CHAR then
+      begin
+        if Uni_IsSurrogate(pc.character) then
+          Inc(Result);
+      end;
+      Inc(Result);
+      Inc(pc);
+    end;
+  end;
+
   function ContextToDebugString(context_items: pkm_core_context_item; offset, length: Integer): string;
   var
-    outText: array of UInt32;
-    ox: Integer;
     pc: pkm_core_context_item;
     dk: TDeadKeyInfo;
   begin
-    SetLength(outText, length + 1);
-
-    ox := 0;
+    Result := '';
```

Comment:
Should this initialize to something with `Result.Length = 0`? Could a loop enter the `else` block on the first pass and hit l.514?

Reply:
`Result := ''` means that `Result.Length = 0`; `.Length` is a calculated property